Makefile

```diff
@@ -1,80 +1,84 @@
-	cargo test -- --test-threads=1
+test:
+	cargo nextest run --lib $(CARGO_FLAGS) --no-fail-fast ${CARGO_BUILD_ARGS}
 	pnpm --recursive test
 
-test-ci: $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT) $(AUTOGENERATED_EMILY_CLIENT)
-	cargo nextest run --no-fail-fast
-	pnpm --recursive test
+test-build:
+	cargo test build $(CARGO_FLAGS) --no-run --locked ${CARGO_BUILD_ARGS}
 
 lint:
-	cargo clippy -- -D warnings
 	cargo fmt --all -- --check
+	cargo clippy -- -D warnings
 	pnpm --recursive run lint
 
-format: $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT) $(AUTOGENERATED_EMILY_CLIENT)
+format:
 	cargo fmt
 
+contracts:
+	pnpm --prefix contracts run build
+
 clean:
 	cargo clean
 	pnpm --recursive clean
 
-.PHONY: install build test lint format clean
+.PHONY: install build test test-build lint format contracts clean
 
-# Integration tests.
-# ------------------------------------------------------------------------------
+# ##############################################################################
+# NEXTEST
+# ##############################################################################
+
+NEXTEST_ARCHIVE_FILE := target/nextest/nextest-archive.tar.zst
+NEXTEST_SERIAL_ARCHIVE_FILE := target/nextest/nextest-archive-serial.tar.zst
+
+# Creates nextest archives
+nextest-archive:
+	cargo nextest archive $(CARGO_FLAGS) --lib --archive-file $(NEXTEST_ARCHIVE_FILE) ${CARGO_BUILD_ARGS}
+	cargo nextest archive $(CARGO_FLAGS) --archive-file $(NEXTEST_SERIAL_ARCHIVE_FILE) --test integration ${CARGO_BUILD_ARGS}
 
-integration-env-up:
-	docker compose --file docker/docker-compose.test.yml up --detach
+# Runs nextest archives
+nextest-archive-run:
+	cargo nextest run --no-fail-fast --retries 2 --archive-file $(NEXTEST_ARCHIVE_FILE)
+	cargo nextest run --no-fail-fast --test-threads 1 --retries 2 --archive-file $(NEXTEST_SERIAL_ARCHIVE_FILE)
 
-integration-test: $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT) $(AUTOGENERATED_EMILY_CLIENT)
-	cargo test --test integration --all-features --no-fail-fast -- --test-threads=1
+nextest-archive-clean:
+	rm -f $(NEXTEST_ARCHIVE_FILE) $(NEXTEST_SERIAL_ARCHIVE_FILE)
 
-integration-test-ci: $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT) $(AUTOGENERATED_EMILY_CLIENT)
-	cargo nextest run --test integration --all-features --no-fail-fast --test-threads=1
+.PHONY: nextest-archive nextest-archive-run nextest-archive-clean
+
+# ##############################################################################
+# INTEGRATION TESTS
+# ##############################################################################
+
+integration-env-up: emily-cdk-synth
+	docker compose --file docker/docker-compose.test.yml up -d
+
+integration-test:
+	cargo nextest run $(CARGO_FLAGS) --test integration --no-fail-fast --test-threads 1
+
+integration-test-build:
+	cargo test build $(CARGO_FLAGS) --test integration --no-run --locked
 
 integration-env-down:
 	docker compose --file docker/docker-compose.test.yml down -v
```

Comment:
nit: can we add a `-t 0` here? I don't think we care about graceful shutdown for this

Reply:
we can[816e2b3](https://github.com/stacks-network/sbtc/pull/1238/commits/816e2b37e0cf5209a25cd5c50313c911d058dc33)
[f8327b8](https://github.com/stacks-network/sbtc/pull/1238/commits/f8327b8d23c94156aa0c8c73050c12f7c7a6536e)
[cb4947b](https://github.com/stacks-network/sbtc/pull/1238/commits/cb4947bb4397208f4f93ef9fd26d1ef4a1b396af)

Went ahead and added it to all of the `down`s with the exception of `devenv-sbtc-down` (since i use this and want it to shutdown gracefully because i generally start it up again).